### ISSUE TITLE
Switch to CLIP Image Embedding for Enhanced Performance

### ISSUE THREAD

**Issue by micedevai** [orpatashnik/StyleCLIP#130]
The current implementation relies on text embeddings for processing visual tasks. However, using CLIP image embeddings instead of text embeddings can significantly enhance performance in tasks such as image comparison, retrieval, and classification. By leveraging CLIP's powerful vision encoder, we can generate embeddings directly from images, improving the relevance and accuracy of image-based tasks.

**Proposal:**

- Replace text embedding-based methods with CLIP image embeddings.
- Utilize CLIP's pre-trained vision model to extract meaningful image features.
- Ensure compatibility with existing workflows by adapting the system to use image embeddings where applicable.

**Steps to Implement:**

1. Install CLIP:  
   ```bash
   pip install git+https://github.com/openai/CLIP.git
   ```

2. Load CLIP and generate image embeddings:
   ```python
   import clip
   import torch
   from PIL import Image

   model, preprocess = clip.load("ViT-B/32", device="cuda" if torch.cuda.is_available() else "cpu")

   image = preprocess(Image.open("path_to_image.jpg")).unsqueeze(0).to(device)
   with torch.no_grad():
       image_features = model.encode_image(image)
   ```

3. Replace text embedding methods with the generated image embeddings in relevant parts of the system.

**Benefits:**

- Direct image embeddings that are better suited for visual tasks.
- Improved performance in image similarity and retrieval.
- Elimination of the need for text-based representations when processing visual data.

---

This issue will help track the transition from text-based embeddings to CLIP's image embeddings and ensure enhanced performance in image-centric tasks.

**Comment by d14945921** [comment]
@micedevai any news?

**Comment by loboere** [comment]
any news?
